boards/arm/nucleo_f429zi/nucleo_f429zi.dts

```diff
@@ -96,3 +97,51 @@ arduino_serial: &usart6 {};
 &iwdg {
 	status = "ok";
 };
+
+&flash0 {
+	/*
+	 * For more information, see:
+	 * http://docs.zephyrproject.org/devices/dts/flash_partitions.html
+	 */
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* 128K for bootloader */
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00020000>;
+			read-only;
+		};
+
+		slot0_partition: partition@20000 {
+			label = "image-0";
+			reg = <0x00020000 0x00040000>;
+		};
+
+		/* backup slot: 256KB */
+		slot1_partition: partition@60000 {
+			label = "image-1";
+			reg = <0x00060000 0x00040000>;
+		};
+
+		/* swap slot: 256KB */
```

Comment:
I thought mcuboot was using only a small slot to do the swap, is 256k really needed?

Reply:
the scratch partition is used to temporarily store the entire data in either slot0 or slot1. So, its size should be the same as that of slot0 or slot1.

Reply:
According to https://github.com/JuulLabs-OSS/mcuboot/blob/master/docs/design.md#image-slots:
`The scratch area must have a size that is enough to store at least the largest sector that is going to be swapped`
And considering the image size, this leaves around ~5000 upgrade cycles, which is more than  enough.